pprof/pprof_test.go

```diff
@@ -28,11 +28,14 @@ func TestPprofServerEnabled(t *testing.T) {
 			require.NoError(t, err)
 		}()
 
+		err := waitForServerReady(url, 5*time.Second)
+		require.NoError(t, err, "Pprof server should be ready")
+
 		req, reqErr := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
```

Comment:
we dont need the additional NewRequestWithContext here, right? waitForServerReady does the exact same thing anyways.

Reply:
I wanted to also check the status code but yeah  `waitForServerReady ` func is better place for that - changed